test/unit/testGoogleHelpers/testPaGetTokens.js

```diff
@@ -0,0 +1,31 @@
+// Third-party dependencies
+const { expect } = require('chai')
+const { describe, it } = require('mocha')
+const chai = require('chai')
+const chaiAsPromised = require('chai-as-promised')
+const rewire = require('rewire')
+
+// In-house dependencies
+const googleHelpers = rewire('../../../lib/googleHelpers')
+const { mockOAuth2Client } = require('../../testingHelpers')
+
+chai.use(chaiAsPromised)
+
+// have googleHelpers use a mock OAuth2Client instead of Google's
+// eslint-disable-next-line no-underscore-dangle
+googleHelpers.__set__('paOAuth2Client', mockOAuth2Client)
+
+describe('googleHelpers.js unit tests: paGetTokens', () => {
+  describe('for an invalid authorization code', () => {
+    it('should throw an Error', () => {
+      expect(googleHelpers.paGetTokens('invalid-authorization-code')).to.be.rejected
```

Comment:
`rejected` is from `chai-as-promised`, which I chose to use because I wasn't entirely sure how to write checks for `async` functions that throw errors... but now that you've recommended `await`ing the function, and checking for `throw()`, that sounds like a much better solution, and also removes the need for `chai-as-promised`. I'll remove that package and check for errors this way! Thanks!

Reply:
I've tried to use the scheme of `expect(await googleHelpers.paGetTokens('invalid-authorization-code')).to.throw()` in place of `expect(googleHelpers.paGetTokens('invalid-authorization-code')).to.be.rejected`, but it hasn't worked. If the function throws an error, then even if it expects an error to be thrown, the test case fails. Please see this StackOverflow discussion for why this happens: https://stackoverflow.com/questions/45466040/verify-that-an-exception-is-thrown-using-mocha-chai-and-async-await .

As said in that discussion, the work around is to create helper functions that wrap the function you want to catch an error for; or, to use `chai-as-promised`'s "to.be.rejected" chain for Promises. I'm going to continue to use `chai-as-promised` because it keeps additional logic out of the test cases, but I'm curious to know if you have other ideas to work around this. (Pending Monday's Time with Theresa meeting!)

Reply:
I'm cool with this. Thanks for looking into it!